integrations/tests/test_shuffle.py

```diff
@@ -0,0 +1,307 @@
+
+sys_args_template = ['/var/ossec/integrations/shuffle.py', '/tmp/shuffle-XXXXXX-XXXXXXX.alert', '',
+                     'http://<IP>:3001/api/v1/hooks/<HOOK_ID>', ' > /dev/null 2>&1']
+
+
+@pytest.mark.parametrize('args', [sys_args_template])
+def test_main_bad_arguments_exits(args):
+    """
+    Test that main function exits when wrong number of arguments are passed
+
+    Parameters
+    ----------
+    args: list[str]
+       list of the arguments passed to the main function
+
+    """
+    with patch("builtins.open", mock_open()), \
+            pytest.raises(SystemExit) as pytest_wrapped_e:
+        shuffle.main(args[0:2])
+    assert pytest_wrapped_e.type == SystemExit
+    assert pytest_wrapped_e.value.code == 2
+
+
+@pytest.mark.parametrize('args', [sys_args_template])
+def test_main_exception(args):
+    """
+    Test exception handling in main when process_args raises an exception
+
+    Parameters
+    ----------
+    args: list[str]
+       list of the arguments passed to the process_args function
+
+    """
+
+    with patch('shuffle.process_args') as process, \
+            patch("builtins.open", mock_open()), \
+            pytest.raises(Exception) as pytest_wrapped_e:
+        process.side_effect = Exception
+        shuffle.main(args)
+
+
+@pytest.mark.parametrize('args, alert', [(sys_args_template, alert_template)])
+def test_main_correct_execution(args, alert):
+    """
+
+    Test the correct execution of the main function
+
+    Parameters
+    ----------
+    args: list[str]
+       list of the arguments passed to the process_args function
+
+    alert: json
+       template alert read from the alert file
+
+    """
+    with patch("builtins.open", mock_open()), \
+            patch('json.load', return_value=alert), \
+            patch('requests.post', return_value=requests.Response):
+        shuffle.main(args)
+
+
+@pytest.mark.parametrize('args', [sys_args_template])
+def test_process_args_alert_file_exit(args):
+    """
+    Test that process_args function exits when alert file is not found
+
+    Parameters
+    ----------
+    args: list[str]
+       list of the arguments passed to the process_args function
+
+    -------
+
+    """
+    with pytest.raises(SystemExit) as pytest_wrapped_e:
+        shuffle.process_args(args)
+    assert pytest_wrapped_e.type == SystemExit
+    assert pytest_wrapped_e.value.code == 3
+
+
+@pytest.mark.parametrize('args, alert', [(sys_args_template, alert_template)])
+def test_process_args(args, alert):
+    """
+    Test the correct execution of the process_args function
+
+    Parameters
+    ----------
+    args: list[str]
+       list of the arguments passed to the process_args function
+
+    alert: json
+       template alert read from the alert file
+
+
+    """
+    with patch("builtins.open", mock_open()), \
+            patch('json.load', return_value=alert), \
+            patch('requests.post', return_value=requests.Response):
+        shuffle.process_args(args)
```

Comment:
To test that the function is working as expected you must ensure that both `generate_msg` and `send_msg` functions are being invoked with the expected parameters.